src/algokit/core/typed_client_generation.py

```diff
@@ -88,6 +102,43 @@ def find_generate_command(self, version: str | None) -> list[str]: ...
     def format_contract_name(self, contract_name: str) -> str:
         return contract_name
 
+    def generate_all(
+        self,
+        app_spec_path_or_dir: Path,
+        output_path_pattern: str | None,
+        *,
+        raise_on_failure: bool,  # TODO: NC - Maybe we should return the error instead?
```

Comment:
Perhaps another option could be to remove raise_on_failure and add dataclass to represent result from generate_all response that will contain success and failure, something like
```@dataclass
    class GenerateAllResult:
        successful_generations: list[tuple[Path, Path]]  # List of (app_spec, output_path) that succeeded
        failed_generations: list[tuple[Path, str]]  # List of (app_spec, error_message) that failed
        
        @property
        def has_failures(self) -> bool:
            """Returns True if any generations failed."""
            return len(self.failed_generations) > 0
 ```
 
 and then the generate.py would be taking care of deciding when to raise an exception